### PR TITLE
Improve registry script usability

### PR DESCRIPTION
**Pull request by karlb**
A large collection of UI improvements for the service registration. Highlights:
* info command to show past deposits
* extend command to prolong registration
* no need to remember your deposit address, `withdraw` will find it for you

Closes https://github.com/raiden-network/raiden-services/issues/768 and https://github.com/raiden-network/raiden-services/issues/490.

### CODE REVIEW

File: src/raiden_libs/service_registry.py [hunk 1]
```diff
@@ -176,15 +190,140 @@ def register(
...
+    # Get required deposit
+    required_deposit = service_registry_contract.functions.currentPrice().call()
+    click.secho(
+        f"\nThe current required deposit is fmt_amount({required_deposit})"
+        "\n\tNote: The required deposit continuously decreases, but "
+        "\n\t      increases significantly after a deposit is made."
+    )
+    if web3.eth.chainId == 1:
+        click.secho("You don't have sufficient tokens", err=True)
+        sys.exit(1)
```
Comment (konradkonrad) [inline src/raiden_libs/service_registry.py:234]: I don't understand the flow here:
a) where is the check that the user does not have "sufficient tokens"?
b) why don't we show the token amount the user has?

File: src/raiden_libs/service_registry.py [hunk 2]
```diff
@@ -360,40 +442,106 @@ def maybe_prompt(query: str) -> None:
...
+        abi=CONTRACT_MANAGER.get_contract_abi(CONTRACT_CUSTOM_TOKEN),
+    )
+    fmt_amount = get_token_formatter(deposit_token_contract)
+
+    # Find deposits
+    deposits = find_deposits(web3, service_address, service_registry_contract, start_block)
+    deposits = [d for d in deposits if not d["withdrawn"]]
+    if not deposits:
+        click.echo("No deposits found!", err=True)
+        sys.exit(1)
```
Comment (konradkonrad) [inline src/raiden_libs/service_registry.py:503]: Not finding withdrawable deposits does not necessarily constitute an error (`exit > 0`) to me. Does it to you?

File: src/raiden_libs/service_registry.py [hunk 3]
```diff
@@ -176,15 +190,140 @@ def register(
...
+
+    It will round to full tokens according to the decimal definition in the
+    token contract, but it will show at least `min_sig_figures` significant
+    figures. This ensures that small token amount stay readable and won't get
+    rounded to zero.
+    """
+    symbol = token_contract.functions.symbol().call()
+    decimals = token_contract.functions.decimals().call()
+
+    def format_token_amount(amount: float) -> str:
```
Comment (konradkonrad) [inline src/raiden_libs/service_registry.py:207]: Neat!

File: src/raiden_libs/service_registry.py (hunk 2 above)
Comment (karlb) [reply src/raiden_libs/service_registry.py:503]: If withdraw is called, I would expect a withdraw to happen. It feels similar to an `rm` with a filename that does not exist to me. I do see your point, but prefer the current return code.

File: src/raiden_libs/service_registry.py (hunk 1 above)
Comment (karlb) [reply src/raiden_libs/service_registry.py]: You are completely right. I moved those lines around a few times too often and apparently mixed up the conditions the last time. Should be better, now.

**Review by konradkonrad — APPROVED** [review]
lgtm, thanks!

**Pull request merged**